vcr/vcr.go

```diff
@@ -142,11 +142,24 @@ func (c *vcr) initIndices() error {
 				var leiaParts []leia.IndexPart
 
 				for _, iParts := range index {
-					name := iParts
-					jsonPath := t.ToVCPath(iParts)
-					leiaParts = append(leiaParts, leia.NewJSONIndexPart(name, jsonPath))
+					name := iParts.ConceptPath
+					indexType := iParts.IndexType
+					jsonPath := t.ToVCPath(iParts.ConceptPath)
+					switch indexType {
+					case concept.IndexTypeBytes:
+						leiaParts = append(leiaParts, leia.NewJSONIndexPart(name, jsonPath, nil, nil))
+					case concept.IndexTypeText:
+						leiaParts = append(leiaParts, leia.NewJSONIndexPart(name, jsonPath, leia.WhiteSpaceTokenizer, concept.CologneTransformer))
+					default:
+						return fmt.Errorf("unknown index type %c @ %s", indexType, name)
+					}
 				}
 
+				// since only new indices trigger a re-index. We'll remove the old one first
+				indexName := fmt.Sprintf("index_%d", i)
+				if err := collection.DropIndex(indexName); err != nil {
```

Comment:
Do I understand correctly we always rebuild the indexes? So if one changes its type or path by accident (or on purpose) next restart everything should be OK again?

Reply:
This is done because we don't have a versioning and migration path yet. Doesn't hurt to rebuild for now